README.md

```diff
@@ -4,128 +4,336 @@ Daniel Ehrenberg, Igalia. Stage 3
+Number(151851850485185185047n)
+// ↪ 151851850485185200000
 
-Silently losing precision sometimes may be a problem, but in most dynamically typed programming languages which provide integers and floats, integers are written like `1` and floats are written like `1.0`. It's possible to scan code for operations which may introduce floating point precision by looking for a decimal point. JavaScript exacerbates the scope of losing precision by making the unfortunate decision that a simple literal like `1` is a float. So, if mixed-precision were allowed, an innocent calculation such as `2n ** 53n + 1` would produce the float `2**53`--defeating the core functionality of this feature.
+```
 
-To avoid this problem, this proposal bans implicit coercions between Numbers and BigInts, including operations which are mixed type. `1n + 1` throws a TypeError. So does passing `1n` as an argument into any JavaScript standard library function or Web API which expects a Number. Instead, to convert between types, an explicit call to `Number()` or `BigInt()` needs to be made to decide which domain to operate in. `0 === 0n` returns `false`.
+### Other Exceptions
 
-Comparisons form an exception to this rule: It's mathematically well-defined to allow comparison operators such as `<` and `==` compare between Numbers and BigInts. Unlike operators like `+`, there is no loss of precision, since the output is just a Boolean. Further, `==` is extended in comparisons with strings in analogous ways, such as `0n == ""` is `true`. Although we may not be able to extend this support to user-defined types, it seems sufficiently important for meeting user expectations that this proposal adds them as a one-off.
+Attempting to convert a fractional value to a `BigInt` throws an exception both when the value is represented as an `Number` and a `String`.
 
-## Design goals
+```js
+BigInt(1.5)
+// ↪ RangeError: The number 1.5 is not a safe integer and thus cannot be converted to a BigInt
 
-### Don't break user intuition
+BigInt('1.5')
+// ↪ SyntaxError: Cannot convert 1.5 to a BigInt
+
+```
+
+`Math.fround` and `|` throw exceptions when called with a `BigInt` to preserve the current expectations of `asm.js`.
+
+```js
+
+Math.fround(1n)
+// ↪ TypeError: Cannot convert a BigInt value to a number
+
+1n|0
+// ↪ TypeError: Cannot mix BigInt and other types, use explicit conversions
+
+```
+
+
+## About the Proposal
+
+### Motivation: Why Do We Need Such Big Numbers?
+
+*NOTE: _These still need to be blown out_*.
+
+- guids
+- fs.stat
+- nanoseconds!
+- ability to store credit cards as numbers
+- include error example with chips and incorrect arithmetic?
+
+### Design Goals, Or Why Is This Like This?
+
+These principles guided the decisions made with this proposal.
+
+
+*NOTE: _These could possibly be edited to be a little clearer for folks with less context or could just be kept in the advanced section_*.
```

Comment:
Agreed that my original text here is not very clear. It might make sense to just leave those design goals in the ADVANCED doc (looks like you copied it there too), and focus here on the intuition of how to use BigInt.

Reply:
What do you think about doing a simpler version here and keeping the more complex ones in ADVANCED? For some kinds of learners, understanding the design why can really help with intuition, so I'd love to keep a little of it here, if you're down?